source/upgrade.txt

```diff
@@ -151,6 +151,11 @@ Version 5.0 Breaking Changes
   responding in the same way as ``Decimal128`` values. In particular,
   ``BsonDecimal128.isNumber()`` now returns ``true``, and
   ``BsonDecimal128.asNumber()`` returns the equivalent ``BsonNumber``.
+.. _java-breaking-changes-v5.0-list-collections:
+
+- This driver version breaks binary compatibility by adding a new
+  ``authorizedCollection`` option to the ``listCollections`` command. This
+  change does not require any changes to source code.
```

Comment:
For a user to know whether his code has to be recompiled due to a change that broke binary compatibility, the user must know whether his code uses the program element that was changed. The current text does not mention that program element (the methods `MongoDatabase.listCollectionNames` were changed in a way that breaks binary compatibility).